lavague-core/lavague/core/extractors.py

```diff
@@ -16,19 +17,32 @@ class JsonFromMarkdownExtractor(BaseExtractor):
     Completion:
     --------------------------------------------
     """
-
-    def extract(self, markdown_text: str) -> str:
+   
+    def extract(self, markdown_text: str, action_shape_validator = None) -> str:
+        # Validate the JSON format
+        
         # Pattern to match the first ```python ``` code block
         pattern = r"```json(.*?)```"
 
         # Using re.DOTALL to make '.' match also newlines
         match = re.search(pattern, markdown_text, re.DOTALL)
-        if match:
-            # Return the first matched group, which is the code inside the ```python ```
-            return match.group(1).strip()
-        else:
-            # Return None if no match is found
-            return None
+        
+        if action_shape_validator:
+            try:
+                #checks if the json returned from the llm matchs the schema
+                validate(instance=json.loads(match.group(1).strip()), schema=action_shape_validator)
+            except json.JSONDecodeError as e:
+                raise(f"Invalid JSON format: {e}")
+            except ValidationError as e:
+                raise(f"JSON does not match schema: {e}")
```

Comment:
Can you raise a custom exception? Something like `ExtractionError`. It will make easier to catch it later